documentation/modules/ROOT/pages/03-demo.adoc

```diff
@@ -7,28 +7,879 @@
+        throw new IllegalArgumentException("Driver does not exist, id = " + driverId);
+    }
+    Address from = addressRepository.getByHash(transitDetails.from.getHash());
+    Address to = addressRepository.getByHash(destinationAddress.getHash());
+    Money finalPrice = completeTransitService.completeTransit(
+      driverId, requestUUID, from, to); // <2>
+    Money driverFee = driverFeeService.calculateDriverFee(
+      finalPrice, driverId);
+    driverService.markNotOccupied(driverId); // <2>
+    transitDetailsFacade.transitCompleted(requestUUID,
+      Instant.now(clock), finalPrice, driverFee); // <2>
+    awardsService.registerMiles(transitDetails.client.getId(),
+      transitDetails.transitId); // <2>
+    invoiceGenerator.generate(finalPrice.toInt(),
+      transitDetails.client.getName() + " " +
+      transitDetails.client.getLastName()); // <2>
+    eventsEventSender.publish(new TransitCompleted(
+      transitDetails.client.getId(), transitDetails.transitId,
+      transitDetails.from.getHash(), destinationAddress.getHash(),
+      transitDetails.started, Instant.now(clock), Instant.now(clock))
+    ); // <2>
+}
+----
+
+[CAUTION]
+====
+There are issues with the above method.
+
+<1> It uses the `+@Transational+` annotation, and modify number of unrelated data stores.
+<2> It merges different, business domains.
+====
+
+Similar methods are, unfortunately, quite common in business applications.
+At first glance, many developers don't see any problems with similar code.
+Let's break down the problems in detail.
+
+==== Overuse of transactional processing
+
+The transactional processing has been the cornerstone of many business applications.
+However, in most cases, the transactional processing isn't the best fit for real-world processes.
+
+In our example, when the ride finishes, that's a real-world situation.
+However, the example uses the `+@Transational+` annotation, and operate on number of unrelated data.
+This means that when one of those operations fails, the whole processing will be rolled back.
+In effect, the end-user will receive a nasty error message.
+
+[NOTE]
+====
+Outages from the dependant services must not invalidate the main intent.
+In fact, all the operations in this example could happen independently, and at different, yet reasonable times.
+====
+
+==== Bundling of different logical domains
+
+Our example is also very chatty, and hard to understand at first glance.
+In fact, this is quite common in similar applications.
+The code starts small, easy to understand.
+When new features are added, it keeps growing as developers cramp new instructions into methods like `+completeTransit+`.
+
+Of course, the developers could re-architect the code, to extract instructions to a separate blocks, but that is just a half-measure.
+Still, the application will do all the operations, starting from `+completeTransit+` method in the same time, and withing the same _"script"_.
+
+=== Refactoring
```

Comment:
you lost me a bit with details here on the refactoring - but I got that smaller modules are better - I think that's perhaps what you wanted to say